src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/jpa/Report.kt

```diff
@@ -253,7 +244,22 @@ class Report(
       addStatusHistory(newStatus, now, updatedBy)
     }
 
-    // TODO: need to compare and update other fields and related entities
```

Comment:
there are still a few fields on `Report` that ought to be copied across.

the slightly harder question is what do we do about `Event`? if the report originally came from NOMIS, the event is a copy of a bunch of report fields – if we know this update is to a report from NOMIS, we should copy across the changes to the event. but what if the original report came from DPS?

Reply:
You answered your own question: I don't think the event should be updated when updating/syncing from NOMIS. The upsert itself doesn't contain any (explicit) information about the event and we only populated it on creation with good enough (guessed) values...but after the Report is created the Event could have changed in the DPS service and we wouldn't want an upsert without explicit event information to reset these potential changes.

Reply:
Just to recap, for now I'll update the event fields that we update. In the future we may or may not break this link on update.

https://github.com/ministryofjustice/hmpps-incident-reporting-api/pull/80/commits/034ee1a850e60738311749476142112959889415